e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/ConstraintsTests.java

```diff
@@ -478,8 +491,39 @@ void compilationFailsOutdatedSimulationNoSimDataset() throws IOException {
       assertEquals(false,constraintRepsonse.success());
       assertEquals(1,constraintRepsonse.errors().size());
       final ConstraintError error = constraintRepsonse.errors().get(0);
-      assertEquals("Constraint 'fruit_equal_peel': TypeError: TS2345 Argument of type '\"/my_boolean\"' is not assignable to parameter of type 'ResourceName'.",error.message());
+      assertEquals("Constraint 'fruit_equal_peel' compilation failed:\n"
+                   + " TypeError: TS2345 Argument of type '\"/my_boolean\"' is not assignable to parameter of type 'ResourceName'.",error.message());
+
+    }
+
+    @Test
+    @DisplayName("if a constraint is updated and then reverted, Constraints will load the cached value from before the update.")
+    void constraintInvalidModification() throws IOException {
+
+      hasura.awaitSimulation(planId);
+      final var constraintsResponses = hasura.checkConstraints(planId);
+
+      assertEquals(1, constraintsResponses.size());
+      assertEquals(true, constraintsResponses.get(0).success());
+
+      hasura.updateConstraint(
+          constraintsResponses.get(0).result().constraintId(),
+          "export default (): Constraint => Discrete.Resource(\"/my_boolean\").equal(true); error");
+
+      final var constraintsErrorResponses = hasura.checkConstraints(planId);
+      assertEquals(1, constraintsErrorResponses.size());
+      assertEquals(false, constraintsErrorResponses.get(0).success());
+      assertEquals(1 , constraintsErrorResponses.get(0).errors().size());
+
+      hasura.updateConstraint(
+          constraintsResponses.get(0).result().constraintId(),
+          "export default (): Constraint => Discrete.Resource(\"/my_boolean\").equal(true)");
+      final var constraintsCacheResponses = hasura.checkConstraints(planId);
+
+      assertEquals(1, constraintsCacheResponses.size());
+      assertEquals(true, constraintsCacheResponses.get(0).success());
 
+      assertEquals(constraintsResponses.get(0),constraintsCacheResponses.get(0));
```

Comment:
This can be simplified to `assertEquals(constraintsResponses, constraintsCacheResponses)`.

Reply:
Simplified as stated above